apps/api/src/persistence/database/arango-query-runner.ts

```diff
@@ -0,0 +1,35 @@
+import { HasId } from '@coscrad/api-interfaces';
+import { Injectable } from '@nestjs/common';
+import { isDeepStrictEqual } from 'util';
+import { ICoscradQueryRunner } from '../migrations/coscrad-query-runner.interface';
+import { ArangoDatabase } from './arango-database';
+import { ArangoCollectionId } from './collection-references/ArangoCollectionId';
+import { ArangoDatabaseProvider } from './database.provider';
+import { DatabaseDocument } from './utilities/mapEntityDTOToDatabaseDTO';
+
+@Injectable()
+export class ArangoQueryRunner implements ICoscradQueryRunner {
+    private readonly arangoDatabase: ArangoDatabase;
+
+    constructor(arangoDatabaseProvider: ArangoDatabaseProvider) {
+        this.arangoDatabase = arangoDatabaseProvider.getDBInstance();
+    }
+
+    async update<TOldDocument extends DatabaseDocument<HasId>, UNewDocument>(
```

Comment:
I think we should constrain `UNewDocument` as well. The update will only work in Arango if the update document has a key \ some ID

Reply:
done